### DISCUSSION TITLE
Why FastAPI has nothing?

### DISCUSSION THREAD

**Discussion by lanforce** [fastapi/fastapi#12006] [Questions] [answered]
I expected better support from such a popular programming language, but it reveals it's mostly empty or abandoned like in other "popular" languages like Golang or Rust.

Why each time I go discover other ecosystems, I always need to get back to fucking Rails or Laravel because of lack of native things otherwise if I decide to continue I need to maintain abandoned spaghetti dependencies or write tons of my own implementations?

Lel.

**Comment by Kludex** [comment, marked as answer]
I didn't get what you said. Could you elaborate?

**Reply by lanforce** [reply]
Why we're forced to reinvent the wheel in Python/FastAPI?

**Reply by mattmess1221** [reply]
FastAPI is a specialized framework that makes very few assumptions on what you want to do. If you don't want to "reinvent the wheel" as you say, Django is a more general framework that has batteries included for most scenarios (templates, login, database, admin dashboard, etc).

**Reply by lanforce** [reply]
Django is more full-fledged but still lacks a lot compared to something like Laravel. And I don't mean the ready features for the web development (like templates or auth) but a framework to manage dependencies and maintain extendable modules across companies for example. Though, Python is probably not the best language for this type of job.

**Reply by mattmess1221** [reply]
Are you talking about [pip](https://pip.pypa.io/en/stable/user_guide/)?

**Reply by sylvainmouquet** [reply]
There are `poetry`, `pdm` or `uv` to manage dependencies.
The developer experience for python developers is better than few years ago. By example, framework like `Pydantic` is a game changer

**Reply by lanforce** [reply]
Guys, you haven't understood what I said. I didn't by any means talk about the package manager. Sorry for disturbing.
